control-plane/src/modules/router.ts

```diff
@@ -360,18 +379,18 @@ export const router = s.router(contract, {
     };
   },
   createDeployment: async (request) => {
-    const owner = await auth.jobOwnerHash(request.headers.authorization);
-    const { clusterId, serviceName } = request.params;
-
-    if (UPLOAD_BUCKET === undefined) {
+    const access = await routingHelpers.validateManagementAccess(request);
+    if (!access) {
       return {
-        status: 501,
+        status: 401,
       };
     }
 
-    if (!owner || owner.clusterId !== clusterId || !owner.cloudEnabled) {
```

Comment:
I am no longer checking `cloudEnabled` as the default provider is `mock` which will have the same effect.
I will remove this from the database.